scripts/create_registry_schema.py

```diff
@@ -0,0 +1,365 @@
+                "version_string",
+                "version_suffix",
+                name="dataset_u_version"
+            ),
+            Index("relative_path", "owner", "owner_type"),
+            {"schema": schema},
+        ),
+    }
+
+    Model = type(class_name, (Base,), {**columns, **meta})
+    return Model
+
+
+def _Dependency(schema, has_production, production="production"):
+    """
+    Links datasets through "dependencies".
+
+    Parameters
+    ----------
+    schema          str      Name of schema we're writing to
+    has_production  boolean  True if this schema refers to production schema
+    production      string   Name of production schema
+    """
+
+    class_name = f"{schema}_dependency"
+
+    # Load columns from `schema.yaml` file
+    columns = _get_column_definitions(schema, "dependency")
+
+    # Remove link to production schema if unneeded.
+    if not has_production:
+        del columns["input_production_id"]
+
+    # Update production schema name
+    else:
+        if production != "production":
+            old_col = columns["input_production_id"]
+            fkey = ForeignKey(f"{production}.dataset.dataset_id")
+            new_input_production_id = Column(old_col.name,
+                                             old_col.type,
+                                             fkey)
+            del columns["input_production_id"]
+            columns["input_production_id"] = new_input_production_id
+
+    # Table metadata
+    meta = {"__tablename__": "dependency",
+            "__table_args__": {"schema": schema}}
+
+    Model = type(class_name, (Base,), {**columns, **meta})
+    return Model
+
+
+# The following should be adjusted whenever there is a change to the structure
+# of the database tables.
+_DB_VERSION_MAJOR = 2
+_DB_VERSION_MINOR = 2
+_DB_VERSION_PATCH = 0
+_DB_VERSION_COMMENT = "Add `location_type` for dataset table"
+
+# Parse command line arguments
+parser = argparse.ArgumentParser(
+    description="""
+Creates dataregistry tables for specified schema and connection information (config)""",
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+)
+parser.add_argument(
+    "--schema",
+    help="name of schema to contain tables. Will be created if it doesn't already exist",
+    default=f"{SCHEMA_VERSION}",
+)
+parser.add_argument(
+    "--production-schema", default="production",
+    help="name of schema containing production tables.",
+)
+parser.add_argument("--config", help="Path to the data registry config file")
+
+args = parser.parse_args()
+schema = args.schema
+prod_schema = args.production_schema
+
+# Connect to database to find out what the backend is
+db_connection = DbConnection(args.config, schema)
+if db_connection.dialect == "sqlite":
+    if schema == "production":
+        raise ValueError("Production not available for sqlite databases")
+    # In fact we don't use schemas at all for sqlite
+    schema = None
+else:
+    if schema != prod_schema:
```

Comment:
Don't worry about matching the 3 versions exactly to be safe?

If not, comment can be changed to say minor version is `>` compared, they dont have to match currently

Reply:
The comment needs to be fixed, the code is what I want; namely that "schema" should be backwards-compatible with "prod_schema"